core/src/main/scala/org/typelevel/paiges/Doc.scala

```diff
@@ -570,8 +570,11 @@ object Doc {
   /**
    * Invariant: generate is referentially transparent
    */
-  private[paiges] case class LazyDoc(private val generate: () => Doc) extends Doc {
+  private[paiges] class LazyDoc(private val generate: () => Doc) extends Doc {
     lazy val evaluated: Doc = generate()
+    override def canEqual(that: Any): Boolean = false
+    override def productElement(n: Int): Any = None
+    override def productArity: Int = 0
```

Comment:
This should be 1 I think.

Reply:
That's what I'm not sure of.  None of these overrides really make sense, so I think this is a square peg-round hole change that is perhaps ill-advised.  For example, I don't want anyone to project the function out with a `Product` operator.  But dropping the `extends Product with Serializable` from `Doc` would make wartremover flip out with all the inferred `Product`s.  

I'll change these as you recommend, but I'm not very comfortable with hacks like this.

Reply:
Well we could remove the with Product from Doc. I think that’s fine since the case classes are all private.